src/main/java/mate/academy/repository/CartItemRepository.java

```diff
@@ -0,0 +1,15 @@
+package mate.academy.repository;
+
+import java.util.Optional;
+import mate.academy.model.CartItem;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface CartItemRepository extends JpaRepository<CartItem, Long> {
+    Optional<CartItem> findByIdAndShoppingCartId(Long itemId, Long cartId);
+
+    Optional<CartItem> findByShoppingCartIdAndBookId(Long cartId, Long bookId);
+
+    void deleteAllByBook_Id(Long bookId);
```

Comment:
Can you do it using cascade?

Reply:
In this case, I need to change Book class, but in the description - the class of the book does not have a list of cart items. So I left a unidirectional relationship. And that's why when I delete a book, I delete all the cartitems here.